myuw/static/js/card/schedule/visual.js

```diff
@@ -22,51 +22,70 @@ var VisualScheduleCard = {
             return;
         }
-        WSData.fetch_course_data_for_term(VisualScheduleCard.term, VisualScheduleCard.render_upon_data, VisualScheduleCard.render_error);
+        WSData.fetch_course_data_for_term(VisualScheduleCard.term, VisualScheduleCard.render_handler, VisualScheduleCard.render_handler);
+        if(myuwFeatureEnabled('instructor_schedule')) {
+            WSData.fetch_instructed_course_data_for_term(VisualScheduleCard.term, VisualScheduleCard.render_handler, VisualScheduleCard.render_handler);
+        }
     },
 
-    _has_all_data: function () {
-        if (WSData.normalized_course_data(VisualScheduleCard.term)) {
-            return true;
+    render_handler: function() {
+        if(VisualScheduleCard._has_all_responses()){
+            var course_data = WSData.normalized_course_data(VisualScheduleCard.term);
+            var instructed_course_data = WSData.normalized_instructed_course_data(VisualScheduleCard.term);
+            if (course_data || instructed_course_data){
+                VisualScheduleCard._render();
+            } else {
+                VisualScheduleCard.render_error();
+            }
         }
-        return false;
-    },
 
-    render_error: function() {
-        // CourseCards displays the message
-        $("#VisualScheduleCard").hide();
     },
 
-    render_upon_data: function(course_index) {
-        if (!VisualScheduleCard._has_all_data()) {
-            return;
+    _has_all_responses: function () {
+        // returns true when both schedule API calls have returned, successful or otherwise
+        var course_data = WSData.normalized_course_data(VisualScheduleCard.term);
+        var instructed_course_data = WSData.normalized_instructed_course_data(VisualScheduleCard.term);
+        var course_err_status = WSData.course_data_error_code(VisualScheduleCard.term);
+        var instructed_course_err_status = WSData.instructed_course_data_error_code(VisualScheduleCard.term);
+
+        var has_all_data = (course_data || course_err_status );
+        if(myuwFeatureEnabled('instructor_schedule')){
+            has_all_data = (has_all_data &&
+            (instructed_course_data || instructed_course_err_status ));
         }
-        VisualScheduleCard._render();
+        return has_all_data;
+    },
+
+    render_error: function() {
+        $("#VisualScheduleCard").hide();
     },
 
     // The course_index will be given when a modal is shown.
     _render: function() {
         var term = VisualScheduleCard.term;
         var course_data = WSData.normalized_course_data(term);
+        var instructed_course_data = WSData.normalized_instructed_course_data(term);
 
-        VisualScheduleCard.render_schedule(course_data, term);
+        $("#VisualScheduleCard").show();
+        VisualScheduleCard.render_schedule(course_data, instructed_course_data, term);
 
-        FinalExamSchedule.render(course_data, term, true);
+//        FinalExamSchedule.render(course_data, term, true);
```

Comment:
Why is this commented out?  There's still a SHOW FINAL EXAM SCHEDULE link at the bottom of the visual schedule.